contracts/origination/OriginationControllerSTIRFRY.sol

```diff
@@ -0,0 +1,268 @@
+{
+    using SafeERC20 for IERC20;
+
+    // ============================================ STATE ==============================================
+
+    IFeeController public immutable feeController;
+    IVaultFactory public immutable vaultFactory;
+
+    /// @notice Mapping from hashed currency pair to whether it is allowed
+    mapping(bytes32 => bool) public stirfryPairs;
+
+    // ========================================== CONSTRUCTOR ===========================================
+
+    /**
+     * @notice Creates a new origination controller STIRFRY contract, as well as initializing
+     *         the origination controller base contract.
+     *
+     * @dev For this controller to work, it needs to be granted the ORIGINATOR_ROLE
+     *      in loan core after deployment.
+     *
+     * @param _originationHelpers           The address of the origination shared storage contract.
+     * @param _loanCore                     The address of the loan core logic of the protocol.
+     * @param _feeController                The address of the fee logic of the protocol.
+     * @param _vaultFactory                 The address of the vault factory.
+     */
+    constructor(
+        address _originationHelpers,
+        address _loanCore,
+        address _feeController,
+        address _vaultFactory
+    ) OriginationControllerBase(_originationHelpers, _loanCore) {
+        if (_feeController == address(0)) revert OCS_ZeroAddress("feeController");
+        if (_vaultFactory == address(0)) revert OCS_ZeroAddress("vaultFactory");
+
+        feeController = IFeeController(_feeController);
+        vaultFactory = IVaultFactory(_vaultFactory);
+    }
+
+    // ======================================= LOAN ORIGINATION =========================================
+
+    /**
+     * @notice Initializes a new STIRFRY loan with Loan Core.
+     *
+     * @notice If item predicates are passed, they are used to verify collateral.
+     *
+     * @dev The caller must be a borrower or lender, or approved by a borrower or lender.
+     * @dev The external signer must be a borrower or lender, or approved by a borrower or lender.
+     * @dev The external signer must come from the opposite side of the loan as the caller.
+     *
+     * @param loanTerms                     The terms agreed by the lender and borrower.
+     * @param stirfryData                   The stirfry data to initialize the loan with.
+     * @param borrower                      Address of the borrower.
+     * @param lender                        Address of the lender.
+     * @param sig                           The loan terms signature, with v, r, s fields, and possible extra data.
+     * @param sigProperties                 Signature nonce and max uses for this nonce.
+     * @param itemPredicates                The predicate rules for the items in the bundle.
+     *
+     * @return loanId                       The unique ID of the new loan.
+     */
+    function initializeStirfryLoan(
```

Comment:
Same comment about loan, stirfry, etc....in my framing, it's more like `initializeSwap`